2021-05-05T21:58:45.5160349+01:00  [INF] Application started. Press Ctrl+C to shut down. (dcaefe54)
2021-05-05T21:58:45.5752782+01:00  [INF] Hosting environment: "Development" (c3307c92)
2021-05-05T21:58:45.5827008+01:00  [INF] Content root path: "C:\Users\Hpuser\source\repos\SecurityAssignment\WebApplication1" (b5d60022)
2021-05-05T21:58:46.2183520+01:00 80000065-0003-fb00-b63f-84710c7967bb [INF] Home index accessed (dd6449db)
2021-05-05T22:01:33.3596798+01:00 80000114-0006-fd00-b63f-84710c7967bb [INF] Home index accessed (dd6449db)
2021-05-05T22:03:30.4829438+01:00  [INF] Application started. Press Ctrl+C to shut down. (dcaefe54)
2021-05-05T22:03:30.5373870+01:00  [INF] Hosting environment: "Development" (c3307c92)
2021-05-05T22:03:30.5437745+01:00  [INF] Content root path: "C:\Users\Hpuser\source\repos\SecurityAssignment\WebApplication1" (b5d60022)
2021-05-05T22:03:31.9653910+01:00 8000001d-0004-fc00-b63f-84710c7967bb [INF] Home index accessed (dd6449db)
2021-05-05T22:05:44.0454321+01:00 8000014d-0006-fe00-b63f-84710c7967bb [INF] Home index accessed (dd6449db)
2021-05-05T22:06:03.3321280+01:00 8000014f-0006-fe00-b63f-84710c7967bb [INF] User logged in. (fdce0db4)
2021-05-05T22:06:03.8832578+01:00 8000006b-0003-fb00-b63f-84710c7967bb [INF] Home index accessed (dd6449db)
2021-05-05T22:09:28.1430068+01:00 80000103-0007-fe00-b63f-84710c7967bb [ERR] An unhandled exception has occurred while executing the request. (48a46595)
Internal.Cryptography.CryptoThrowHelper+WindowsCryptographicException: The parameter is incorrect.
   at Internal.NativeCrypto.CapiHelper.DecryptKey(SafeKeyHandle safeKeyHandle, Byte[] encryptedData, Int32 encryptedDataLength, Boolean fOAEP, Byte[]& decryptedData)
   at System.Security.Cryptography.RSACryptoServiceProvider.Decrypt(Byte[] rgb, Boolean fOAEP)
   at System.Security.Cryptography.RSACryptoServiceProvider.Decrypt(Byte[] data, RSAEncryptionPadding padding)
   at WebApplication1.Utility.Encryption.AsymmetricDecrypt(Byte[] cipher, String privateKey) in C:\Users\Hpuser\source\repos\SecurityAssignment\WebApplication1\Utility\Encryption.cs:line 165
   at WebApplication1.Utility.Encryption.HybridDecrypt(MemoryStream encryptedFile, String privateKey) in C:\Users\Hpuser\source\repos\SecurityAssignment\WebApplication1\Utility\Encryption.cs:line 203
   at WebApplication1.Controllers.AssignmentsController.Download(Guid assignmentId) in C:\Users\Hpuser\source\repos\SecurityAssignment\WebApplication1\Controllers\AssignmentsController.cs:line 206
   at lambda_method(Closure , Object , Object[] )
   at Microsoft.Extensions.Internal.ObjectMethodExecutor.Execute(Object target, Object[] parameters)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.MigrationsEndPointMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.DatabaseErrorPageMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.DatabaseErrorPageMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
2021-05-05T22:27:46.5816041+01:00  [INF] Application started. Press Ctrl+C to shut down. (dcaefe54)
2021-05-05T22:27:46.6441867+01:00  [INF] Hosting environment: "Development" (c3307c92)
2021-05-05T22:27:46.6514216+01:00  [INF] Content root path: "C:\Users\Hpuser\source\repos\SecurityAssignment\WebApplication1" (b5d60022)
2021-05-05T22:27:47.7136284+01:00 80000156-0001-fe00-b63f-84710c7967bb [INF] Home index accessed (dd6449db)
